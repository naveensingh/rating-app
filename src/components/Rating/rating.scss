.rating {
  display: inline-block;

  &--star {
    display: inline-block;
    padding: 3px;
    vertical-align: middle;
    line-height: 1;
    font-size: 1.5em;
    color: #ABABAB;
    transition: color .2s ease-out;
    cursor: pointer;
  }
}

.rating:hover .rating--star {
  color: #FFD700;
}

.rating .rating--star:hover ~ .rating--star {
  color: #ABABAB;
}

.rating--star.active {
  color: #FFD700;
}
