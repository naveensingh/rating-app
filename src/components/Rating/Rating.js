import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './rating.scss';
import { MAX_RATING } from '../../constants/Constants';

class Rating extends Component {
    maxRating = MAX_RATING;

    render() {
      const { rating, onClick } = this.props;
      return (
        <div className="rating">
          {
                    [...Array(this.maxRating).keys()].map(key => (
                      <div
                        key={key}
                        className={`rating--star ${classNames({ active: (key + 1) <= rating })}`}
                        onClick={() => onClick(key + 1)}
                      >â˜…</div>
                    ))
                }
        </div>
      );
    }
}

Rating.propTypes = {
  rating: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Rating;
