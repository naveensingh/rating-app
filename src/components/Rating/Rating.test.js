import React from 'react';
import Rating from "./Rating";
import {shallow} from "enzyme";

describe('Rating spec', () => {
    const onclickMock = jest.fn();
    test('should match snapshot for rating', () => {
        const ratingComponent = shallow(<Rating rating={3} onClick={onclickMock}/>);
        expect(ratingComponent).toMatchSnapshot();
    });

    test('if only 3 stars are active', () => {
        const ratingComponent = shallow(<Rating rating={3} onClick={onclickMock}/>);
        expect(ratingComponent.find('.rating--star.active')).toHaveLength(3);
    });

    test('if each star is called with respective rating', () => {
        const ratingComponent = shallow(<Rating rating={3} onClick={onclickMock}/>);
        ratingComponent.find('.rating--star').forEach((node, index) => {
            node.simulate('click');
            expect(onclickMock).toHaveBeenCalledWith(index + 1);
        });
    });
});
