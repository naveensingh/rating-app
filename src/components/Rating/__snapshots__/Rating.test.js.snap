// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating spec should match snapshot for rating 1`] = `
<div
  className="rating"
>
  <div
    className="rating--star active"
    key="0"
    onClick={[Function]}
  >
    ★
  </div>
  <div
    className="rating--star active"
    key="1"
    onClick={[Function]}
  >
    ★
  </div>
  <div
    className="rating--star active"
    key="2"
    onClick={[Function]}
  >
    ★
  </div>
  <div
    className="rating--star "
    key="3"
    onClick={[Function]}
  >
    ★
  </div>
  <div
    className="rating--star "
    key="4"
    onClick={[Function]}
  >
    ★
  </div>
</div>
`;
